### DISCUSSION TITLE
poll: Do you have any interest in smart home monitoring?

### DISCUSSION THREAD

**Discussion by TwiN** [TwiN/gatus#795] [Polls]
For a bit of context, I am going to be moving very soon and given that what led to me creating Gatus in the first place was my love for automation, I am definitely going to try to turn my home into a smart home.

Where I think Gatus could be interesting is when it comes to smart home sensors, such as water sensors (for sump pump, sinks, etc), temperature sensors, humidity sensors, maybe even motion sensors linked to security systems (e.g. outdoor motion sensors or camera movement detection being triggered near the front door at night).

I have no idea _how_ to pull detected events into Gatus, but still, now that we have [External Endpoints](https://github.com/TwiN/gatus#external-endpoints), I think pretty much everything is possible. Even if it isn't, I'm sure it's possible to make a few PRs to popular smart home solutions such as Home Assistant & make it possible.

What I'm creating this poll for is to gauge interest in this kind of feature (smart home monitoring). I am fully aware that most people using Gatus do so for work-related reasons, and for those of you, smart home monitoring is completely useless, but there's also quite a lot of people using this to monitor their home infrastructure. For those, turning "home infrastructure monitoring" into "home monitoring" _could_ sound interesting, so, what do you think?

**Comment by Rindula** [comment]
In theory, when using Homeassistant (HA), smarthome monitoring is already possible. HA has the possibility to query sensordata via a rest endpoint using long living api tokens as authentication header. When a sensor becomes unavailable, the state in HA changes to "unavailable" or "unknown", so you could make a simple request checking for that. So I don't think that any further changes to that direction are nessesary.

**Reply by spencerhughes** [reply]
I'm not intimately familiar with the particular capabilities of Home Assistant integrations, but do you think it would be useful if there was a Gatus integration that attempted to discover and auto-create endpoints for your various sensors, so that you could plumb the Home Assistant state-awareness into your existing Gatus infrastructure? Basically architected like a remote Gatus, except it's running as an integration in Home Assistant and does a lot of auto-config for you, based on that context.

**Reply by Rindula** [reply]
This would definetly make things easier. In HA you have different devices with possibly many entities. So in theory you would need to select certain entities and enter their desired state. As long as that state is the same, gatus could assume the endpoint is healthy, if there is any other state the endpoint would be unhealthy (this could also be negated, so only one state is invalid and everything else is valid).
